app/models/recipe.rb

```diff
@@ -1,5 +1,13 @@
 class Recipe < ApplicationRecord
   validates :name, presence: true
+  has_and_belongs_to_many :categories
+  accepts_nested_attributes_for :categories
+
+  def save_categories
+    categories.map do |category|
+      Category.find_or_create_by(name: category.name.strip)
+    end
+  end
```

Comment:
I'm not sure if this is doing what you think it might be.  The `categories` it is accessing now would be the categories that already exist on the recipe instance, which could possibly work to persist new instances of Categories that haven't been saved to the database yet. But there was nothing initializing the categories. Also, this isn't saving the recipe itself, so it can't replace the save method.